OSDOCS-19753: adds Vault integration MCP gateway (#111554)
openshift/openshift-docs: modules/proc-mcp-gateway-trust-vault-authorino.adoc

[NOTE]
====
In production environments, using a tool such as Terraform or GitOps to manage policies is recommended. Automation ensures that if a cluster is deleted, then your security policies are automatically recreated.
====

.Prerequisites

* You completed the installation for {mcpg}.
* You created a `Gateway` object.
* You have a Vault database.
* You created a namespace for your MCP server secrets.

.Procedure

. Enable JWT authentication in your Vault server by following the instructions in the link:https://developer.hashicorp.com/vault/api-docs/auth/jwt#configure[Vault documentation].

. Create a Vault policy that saves the Access Control List (ACL) that defines the `read` and `list` capabilities for Authorino by running the following command:
+
[source,terminal,subs="+quotes"]
----
$ curl -H "X-Vault-Token: $VAULT_TOKEN" -H 'Content-Type: application/json' -X POST \
  --data '{
    "policy": "path \_<"secret/data/mcp-gateway/*\">_ {\n  capabilities = [\"read\", \"list\"]\n}"
  }' \

>   _<http://vault.vault-namespace.svc.cluster.local:8200>_/v1/sys/policies/acl/authorino

Reviewer: This is curling from outside the cluster, so `http://vault.vault-namespace.svc.cluster.local` can be a misleading example. Maybe use a more obvious placeholder such as `<your-vault-service-hostname>`?

----
+
* Replace `_<"secret/data/mcp-gateway/*\">_` with the path to the namespace for your MCP server secrets.
* Replace `_<\http://vault.vault-namespace.svc.cluster.local:8200>_` with the Vault service address.

. Create a Vault role that stores a set of rules for Authorino by running the following command:
+
[source,terminal]
----
$ curl -H "X-Vault-Token: $VAULT_TOKEN" -H 'Content-Type: application/json' -X POST \
  --data '{
    "role_type": "jwt",
    "bound_audiences": ["authorino"],
    "user_claim": "sub",
    "policies": ["authorino"],
    "ttl": "1h"

>   }'

Reviewer: Missing the URL.

// Module included in the following assemblies:
//
// *mcp_gateway_config/mcp-gateway-vault.adoc

:_mod-docs-content-type: PROCEDURE
[id="proc-mcp-gateway-trust-vault-authorino_{context}"]
= Setting up a trust relationship between Vault and Authorino

[role="_abstract"]
To ensure that the Authorino service can act on an `AuthPolicy` custom resource (CR), you must set up a secure, identity-based trust relationship between the Authorino service and your HashiCorp Vault (Vault) database. Your goal is to allow the Authorino service to dynamically retrieve sensitive credentials, such as API keys or OIDC client secrets, from Vault.

[NOTE]
====
In production environments, using a tool such as Terraform or GitOps to manage policies is recommended. Automation ensures that if a cluster is deleted, then your security policies are automatically recreated.
====

.Prerequisites

* You completed the installation for {mcpg}.
* You created a `Gateway` object.
* You have a Vault database.
* You created a namespace for your MCP server secrets.

.Procedure

> . Enable JWT authentication in your Vault server by following the instructions in the link:https://developer.hashicorp.com/vault/api-docs/auth/jwt#configure[Vault documentation].

Reviewer 2: Move link to add resources section? Consider stating the link is in the Additional resources section.

. Create a Vault policy that saves the Access Control List (ACL) that defines the `read` and `list` capabilities for Authorino by running the following command:
+
[source,terminal,subs="+quotes"]
----
$ curl -H "X-Vault-Token: $VAULT_TOKEN" -H 'Content-Type: application/json' -X POST \
  --data '{
    "policy": "path \_<"secret/data/mcp-gateway/*\">_ {\n  capabilities = [\"read\", \"list\"]\n}"
  }' \
  _<vault_service_hostname:port>_/v1/sys/policies/acl/authorino
----
+
* Replace `_<"secret/data/mcp-gateway/*\">_` with the path to the namespace for your MCP server secrets.
* Replace `_<vault_service_hostname:port>_` with the Vault service address.

. Create a Vault role that stores a set of rules for Authorino by running the following command:
+
[source,terminal,subs="+quotes"]
----
$ curl -H "X-Vault-Token: $VAULT_TOKEN" \
  -H 'Content-Type: application/json' \
  -X POST \
  --data '{
    "role_type": "jwt",
    "bound_audiences": ["authorino"],
    "user_claim": "sub",
    "policies": ["authorino"],
    "ttl": "1h"
  }' \

>   _<vault_service_url:port>/v1/auth/jwt/role/authorino-role>_

Reviewer: Do I see a trailing `>` character that should not be there or am I just misreading the placeholder?

```suggestion
  _<vault_service_url:port>/v1/auth/jwt/role/authorino-role_
```

Author: Yes, looks like this should just be the url and port (consistent with the previous step).

----
+

> Replace `_<vault_service_hostname:port>_` with the Vault service address.

Reviewer: The placeholder here looks inconsistent with line 56.

> Replace `_<"secret/data/mcp-gateway/*\">_` with the URL and port of the your Vault endpoint.

Reviewer: ```suggestion
Replace <vault_service_hostname:port> with the Vault service address.
```
